src/layout.rs

```diff
@@ -114,8 +118,7 @@ impl DisplayLeaf for StringLeaf {
         leaf: u32,
         cpuid: &CPUIDFunc,
     ) -> Vec<CpuidResult> {
-        let cpuid = cpuid.get_cpuid(leaf, 0);
-        if !is_empty_leaf(&cpuid) {
+        if let Some(cpuid) = cpuid.get_cpuid(leaf, 0) {
             vec![cpuid]
         } else {
             vec![]
```

Comment:
could this function that is an if let/else be more idiomatically expressed with a match?
```rust
match cpuid.get_cpuid(leaf, 0) {
    Some(cpuid) => vec![cpuid],
    None => vec![]
}
```
it would probably be good to also disambiguate the names for the input and output cpuid's